presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQuery.java

```diff
@@ -202,6 +202,15 @@ private void waitForMinimumWorkers()
         addExceptionCallback(minimumWorkerFuture, throwable -> queryExecutor.execute(() -> fail(throwable)));
     }
 
+    private void waitForMinimumCoordinatorSidecarsAndWorkers()
+    {
+        ListenableFuture<?> minimumCoordinatorSidecarFuture = clusterSizeMonitor.waitForMinimumCoordinatorSidecars();
+        // when coordinator sidecar requirement is met, start checking for the worker requirement
+        addSuccessCallback(minimumCoordinatorSidecarFuture, this::waitForMinimumWorkers);
+
+        addExceptionCallback(minimumCoordinatorSidecarFuture, this::fail);
+    }
```

Comment:
And also remove `waitForMinimumWorkers`

```suggestion
    private void waitForMinimumCoordinatorSidecarsAndWorkers()
    {
        ListenableFuture<?> futures = Futures.allAsList(
            clusterSizeMonitor.waitForMinimumCoordinatorSidecars(),
            clusterSizeMonitor.waitForMinimumWorkers());

        // when worker and sidecar requirement is met, wait for query execution to finish construction and then start the execution
        addSuccessCallback(minimumWorkerFuture, () -> {
            // It's the time to end waiting for resources
            boolean isDispatching = stateMachine.transitionToDispatching();
            addSuccessCallback(queryExecutionFuture, queryExecution -> startExecution(queryExecution, isDispatching));
        });
        addExceptionCallback(minimumWorkerFuture, throwable -> queryExecutor.execute(() -> fail(throwable)));
    }
```